### ISSUE TITLE
biome ci crashes with SIGABRT: Resource temporarily unavailable

### ISSUE THREAD

**Issue by ekfriis** [biomejs/biome#9947]
## Environment

- **Biome version**: 2.4.10 (config schema 2.3.15)
- **OS**: Linux 6.6.87.2-microsoft-standard-WSL2 (Ubuntu on WSL2)

## Description

`biome ci .` intermittently crashes with `SIGABRT` during linting. The crash is non-deterministic — the same command succeeds on retry, sometimes multiple times in a row, then crashes again.

## Error output

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_console/src/lib.rs:159:14
Thread Name: main
Message: called `Result::unwrap()` on an `Err` value: Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace
```

The shell reports exit code 134 (SIGABRT).

## Reproduction

The crash is intermittent and hard to reproduce reliably. In our CI pipeline, `biome ci .` (checking 272 files) crashes roughly 1 in 3 runs. Running the exact same command immediately after a crash often succeeds.

The codebase includes TypeScript, Svelte, CSS, and JSON files. The biome.json config schema version (2.3.15) does not match the CLI version (2.4.10), which biome warns about but should not cause a crash.

## Expected behavior

`biome ci .` should complete without crashing, or fail gracefully with an error message instead of aborting.

## Additional context

The crash originates in `biome_console/src/lib.rs:159` where an `unwrap()` on a write operation hits `WouldBlock` (EAGAIN). This suggests a non-blocking I/O issue when writing to stdout/stderr, possibly a race condition in console output.

**Comment by dyc3** [comment]
Very weird, we've never seen this be a problem, as far as I know. It could have something to do with WSL?
Could you try this preview build and see if that fixes it?

```
npm i https://pkg.pr.new/@biomejs/biome@9c15f55
```

In this build, I implemented a simple retry when those errors are encountered.

**Comment by dyc3** [comment]
@ekfriis Did the preview build resolve the issue?
